pkg/burner/create.go

```diff
@@ -123,18 +128,29 @@ func (ex *Executor) RunCreateJob(iterationStart, iterationEnd int) {
 		}
 		log.Debugf("Creating object replicas from iteration %d", i)
 		if ex.Config.NamespacedIterations {
-			ns = fmt.Sprintf("%s-%d", ex.Config.Namespace, i)
-			if err = createNamespace(ClientSet, ns, nsLabels); err != nil {
-				log.Error(err.Error())
-				continue
+			ns = fmt.Sprintf("%s-%d", ex.Config.Namespace, i/ex.Config.IterationsPerNamespace)
+			if !namespacesCreated[ns] {
+				if err = createNamespace(ClientSet, ns, nsLabels); err != nil {
+					log.Error(err.Error())
+					continue
+				}
+				namespacesCreated[ns] = true
 			}
 		}
 		for objectIndex, obj := range ex.objects {
 			ex.replicaHandler(objectIndex, obj, ns, i, &wg)
 		}
 		if ex.Config.PodWait {
```

Comment:
Can we condense this block like this, please?
```suggestion
		if ex.Config.PodWait {
                        if !ex.Config.NamespacedIterations || !namespacesWaited[ns] {
                            log.Infof("Waiting up to %s for actions to be completed in namespace", ex.Config.MaxWaitTimeout, ns)
                            ex.waitForObjects(ns)
                            namespacesWaited[ns] = true
                        }
                }
```